install_scripts/templates/kubernetes-upgrade.sh

```diff
@@ -0,0 +1,164 @@
+#!/bin/bash
+
+set -e
+AIRGAP=0
+
+{% include 'common/common.sh' %}
+{% include 'common/kubernetes.sh' %}
+{% include 'common/log.sh' %}
+{% include 'common/prompt.sh' %}
+{% include 'common/system.sh' %}
+
+#######################################
+# Upgrade Kubernetes on remote workers to master version
+# Globals:
+#   AIRGAP
+# Arguments:
+#   None
+# Returns:
+#   None
+#######################################
+upgradeK8sWorkers() {
+    masterVersion=$(getK8sMasterVersion)
+    version="v${masterVersion}"
+
+    workers=$(kubectl get nodes | sed '1d' | grep -v master | grep -v $version | awk '{ print $1 }')
+
+    for node in $workers; do
+        kubectl drain "$node" --ignore-daemonsets --delete-local-data --force
+        printf "\n\n\tRun the upgrade script on remote node before proceeding: ${GREEN}$node${NC}\n\n"
+        if [ "$AIRGAP" = "1" ]; then
+            printf "\t${GREEN}cat kubernetes-node-upgrade | sudo bash -s kubernetes-version=$version${NC}"
+        else
+            printf "\t${GREEN}curl https://get.replicated.com/kubernetes-node-upgrade | sudo bash -s kubernetes-version=$version${NC}"
+        fi
+        printf "\n\n\tContinue when script has completed\n\n"
+        prompt
+        kubectl uncordon $node
+    done
+
+    spinnerNodesReady
+}
+
+#######################################
+# Upgrade Kubernetes on master node
+# Globals:
+#   LSB_DIST
+#   DIST_VERSION
+# Arguments:
+#   k8sVersion - e.g. v1.10.5
+# Returns:
+#   None
+#######################################
+upgradeK8sMaster() {
+    k8sVersion=$1
+    pkgTag=$(k8sPackageTag $k8sVersion)
+
+    if [ "$AIRGAP" = "1" ]; then
+        docker load < packages-kubernetes-${pkgTag}
+    fi
+    docker run \
+      -v $PWD:/out \
+      "quay.io/replicated/k8s-packages:${pkgTag}"
+
+    # must use kubeadm binary to begin upgrade before upgrading kubeadm package
+    # https://kubernetes.io/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-1-11/
+    cp archives/kubeadm /usr/bin/kubeadm
+    chmod a+rx /usr/bin/kubeadm
+
+    # The coredns deployment sets allowPrivilegeEscalation: false, which causes
+    # a crash loop backoff if SELinux is enabled, which is common for CentOS
+    # even if it has been set to permissive mode. Required policy would be:
+    # typebounds container_runtime_t svirt_lxc_net_t;
+    kubeadm upgrade apply $k8sVersion --yes \
+        --config=/opt/replicated/kubeadm.conf \
+        --feature-gates=CoreDNS=false
+
+    # upgrade master
+    master=$(kubectl get nodes | grep master | awk '{ print $1 }')
+    # ignore error about unmanaged pods
+    kubectl drain $master --ignore-daemonsets --delete-local-data 2>/dev/null || :
```

Comment:
Will this render the replicated pods unschedulable? (I think thats okay, since the app should stay running, right?

Reply:
Yes, Replicated will be unavailable. The app might be available if they have enough space on remote worker nodes to continue running it.